cachito/web/models.py

```diff
@@ -674,7 +674,9 @@ def _validate_gitsubmodule_exclusivity(pkg_manager_paths, mutually_exclusive):
         if any(path != "." for path in paths):
             msg = (
                 f"Cannot process non-root packages with '{pkg_manager}' "
-                "when 'git-submodule' is also set"
+                "when 'git-submodule' is also set.\nSolution: remove the git submodules"
+                "and instead process the upstream repos (corresponding to the submodules) directly."
+                "Each repo should be processed in a separate request."
```

Comment:
```suggestion
                "when 'git-submodule' is also set.\nSolution: remove the git submodules "
                "and instead process the upstream repos (corresponding to the submodules) directly. "
                "Each repo should be processed in a separate request."
```
Missing two spaces. You can check the error message with
```shell
curl localhost:8080/api/v1/requests \
    -X POST \
    -H 'content-type: application/json' \
    -d '{
  "repo": "https://whatever.org",                                
  "ref": "b470410e50caff0447bc18ca4f011663681e7e17",
  "pkg_managers": ["git-submodule", "gomod"],
  "packages": {"gomod": [{"path": "foo"}]}
}'
```